prometheus.lua

```diff
@@ -355,6 +403,31 @@ function Prometheus:inc(name, label_names, label_values, value)
   self:log_error_kv(key, value, err)
 end
 
+-- Set the current value of a gauge to `value`
+-- Args:
+--   name: (string) short metric name without any labels.
+--   label_names: (array) a list of label keys.
+--   label_values: (array) a list of label values.
+--   value: (number) the new value for the gauge.
+function Prometheus:set_gauge(name, label_names, label_values, value)
+  local key = full_metric_name(name, label_names, label_values)
+
+  local success, err = self.dict:set(key, value)
```

Comment:
Since for gauges we don't care whether there is an existing value or not, you can just call `self:set_key()` immediately after resolving the `key`.